wfsim/core.py

```diff
@@ -291,89 +291,106 @@ def __call__(self, instruction):
             np.array(v).reshape(-1) for v in zip(*instruction)]
 
         positions = np.array([x, y, z]).T  # For map interpolation
-        if self.config['detector']=='xenonnt_detector':
-            ly = np.squeeze(self.resource.s1_light_yield_map(positions),
-                            axis=-1)
-        elif self.config['detector']=='XENON1T':
-            ly = self.resource.s1_light_yield_map(positions)
-            ly *= self.config['s1_detection_efficiency']
-        n_photons = np.random.binomial(n=n_photons, p=ly)
+        n_photons = self.get_n_photons(n_photons=n_photons,
+                                        positions=positions,
+                                        s1_light_yield_map=self.resource.s1_light_yield_map,
+                                        config=self.config)
 
-        self._photon_timings = np.array([])
-        list(map(self.photon_timings, t, n_photons, recoil_type))
+        self._photon_timings = self.photon_timings(t, n_photons, recoil_type, self.config)
         # The new way iterpolation is written always require a list
-        self.photon_channels(positions, n_photons)
+        self._photon_channels = self.photon_channels(positions, n_photons, self.config, self.resource.s1_pattern_map)
 
         super().__call__()
 
-    def photon_channels(self, points, n_photons):
+    @staticmethod
+    def get_n_photons(n_photons,positions, s1_light_yield_map, config):
+        if config['detector']=='xenonnt_detector':
+            ly = np.squeeze(s1_light_yield_map(positions),
+                            axis=-1)
+        elif config['detector']=='XENON1T':
+            ly = s1_light_yield_map(positions)
+            ly *= config['s1_detection_efficiency']
+        n_photons = np.random.binomial(n=n_photons, p=ly)
+        return n_photons
 
-        # print(self.config)
-        channels = np.arange(self.config['n_tpc_pmts'])#+1 for the channel map
-        p_per_channel = self.resource.s1_pattern_map(points)
-        p_per_channel[:, np.in1d(channels, self.turned_off_pmts)] = 0
+    @staticmethod
+    def photon_channels(points, n_photons,config,s1_pattern_map):
```

Comment:
Just to make the function definition consistent with the other functions, a space should be added in front of `config ` and `s1_pattern_map `.